src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -603,8 +610,21 @@ where
             vers
         };
         let dep = Dependency::parse_no_deprecated(name, vers_spec, source.source_id())?;
-        let deps = source.query_vec(&dep)?;
-        match deps.iter().map(|p| p.package_id()).max() {
+        let deps = (|| {
+            if let Some(vers_spec) = vers_spec {
+                if semver::VersionReq::parse(vers_spec).unwrap().is_exact() {
```

Comment:
Instead of parsing this again, could the requirement through `Dependency` be extracted?